### ISSUE TITLE
Switch from `execution` centric to `run` centric approach

### ISSUE THREAD

**Issue by supersergiy** [ZettaAI/zetta_utils#596]
### Motivation
The existing mechanism for tracking resources focuses on executions, a concept closely tied to `mazepa`. This approach results in unwanted complexity and repeated logic. Additionally, this system poses challenges for advanced tools that aim to initiate and monitor the progress of zutils runs.

### Proposed Solution
Track `run`'s, not `execution`'s. Delegate ID generation and hearbeat logic to `zetta run`. The `RUN_ID` will be made accessible through a global variable: 
```python 
from zetta_utils import RUN_ID
```

Introduce a `Run DB`, which is an analog of the current `Execution DB` and will hold information about each run. More specifically, `Run DB` will contain user, cli arguments, timestamps, status (eg `RUNNING`, `TIMED OUT`, `FAILED`, `SUCCEEDED`), spec files. By delegating such logging to `zetta run` we will be able to log the original `.cue` file, instead of just logging the unpacked `json`. `zetta run` will create a `Run DB` entry, then run the spec inside a try-except clause, and then update the status to `FAILED`/`SUCCEEDED` depending on whether an exception has been thrown. Garbage collector will set the status to `TIMED OUT` when encountering stale heartbeat.

An external tool may want to launch a `zetta run` and track its completion status by querying the `Run DB`. To facilitate that, we should allow generating the `run_id` externally and passing it through cli:
```bash
zetta run --run_id {my-run-id}
```

Sometimes we may use `zetta run` to start auxiliary processes that logically pertain to an existing run, such as `mazepa` and DDP workers. In these cases we can use the `--run_id` arg, but also need an additional `--no_heartbeat` flag to prevent the auxiliary process from sending heartbeats.

### Notes
* Resource tracking primitives should still not use the global `RUN_ID` but rather accept it as an argument to maximize flexibility and minimize assumptions

### Questions
* Is the `--no_heartbeat` flag too easy to forget, causing challenging bugs?
* Which databases/schemas need to be updated?
* Is {`RUNNING`, `TIMED OUT`, `FAILED`, `SUCCEEDED`} a good set of statuses? Are there alternatives?
* Is it a good idea to check whether a run ID already exists in the DB before starting the run?

**Comment by dodamih** [comment]
I think this is a good idea. How would bugs manifest with `--no_heartbeat`?

**Comment by supersergiy** [comment]
If somebody spawns an auxiliary process that emits heartbeats, garbage detector will not be able to detect failures of the main process responsible for the run

**Comment by akhileshh** [comment]
Fixed by #618
